hxckr-org/hxckr-k8s-bucket
# feat: added ingress controller

# Add Ingress Configuration and TLS Setup for Production

## Summary
This PR introduces Ingress configuration for both development and production environments, along with TLS certificate setup using cert-manager for the production environment. These changes improve the application's accessibility and security in Kubernetes deployments.

## Changes

### Ingress Configuration
- Added `ingress.yaml` for both development and production environments
- Updated `kustomization.yaml` files to include the new Ingress resources
- Configured the Ingress to route traffic to the server service

### TLS Certificate Setup
- Added `cert-manager.yaml` to set up a ClusterIssuer for automatic TLS certificate provisioning
- Configured the production Ingress to use cert-manager for TLS

### Documentation
- Updated `README.md` with instructions for Ingress configuration and TLS certificate setup
- Added explanations for DNS configuration and domain customization

### Minor Changes
- Updated `.gitignore` to exclude `.vscode` directory
- Removed trailing newline in `k8s/base/kustomization.yaml`
- Removed comment in `k8s/base/softserve.yaml`

## How to Test
1. Deploy the application to the development environment and verify that the Ingress routes traffic correctly to the server service.

## Notes
- Remember to replace `your-domain.com` in the production Ingress configuration with your actual domain.
- Update the email address in `cert-manager.yaml` for the Let's Encrypt account.

This PR enhances our Kubernetes setup by providing a more robust and secure way to access our application in both development and production environments.
